src/crypto/qat/qcccrypto.h

```diff
@@ -18,27 +30,41 @@ extern "C" {
 }
 
 class QccCrypto {
+    friend class QatCrypto;
+    size_t chunk_size;
+    size_t max_requests;
+
+    boost::asio::io_context my_context;
+    std::thread qat_context_thread;
+    void my_context_run();
+    using work_guard_type = boost::asio::executor_work_guard<boost::asio::io_context::executor_type>;
+    std::unique_ptr<work_guard_type> work_guard;
+
+    std::queue<std::function<void(int)>> instance_completions;
```

Comment:
i think `std::queue` with its default backing of `std::deque` is a little heavy-weight here

since we'll know the `max_queue_size` once `init()` is called, we should use a container that can `reserve()` all of this memory up front. i like [`boost::circular_buffer<T>`](https://www.boost.org/doc/libs/1_79_0/doc/html/boost/circular_buffer.html) for this kind of fifo queue

you might consider replacing `std::queue<int> open_instances;` below as well

Reply:
> since we'll know the `max_queue_size` once `init()` is called, we should use a container that can `reserve()` all of this memory up front. i like [`boost::circular_buffer<T>`](https://www.boost.org/doc/libs/1_79_0/doc/html/boost/circular_buffer.html) for this kind of fifo queue
> 
> you might consider replacing `std::queue<int> open_instances;` below as well

I have changed `instance_completions` and `open_instances` type from `std::queue<T>` to `boost::circular_buffer<T>`.